Shared/Supporting Files/Views/SampleLink.swift

```diff
@@ -0,0 +1,144 @@
+        let description: AttributedString
+        
+        /// A Boolean value indicating whether the sample's description is showing.
+        @State private var isShowingDescription = false
+        
+        /// The names of the favorited samples loaded from user defaults.
+        @AppStorage(UserDefaults.favoritedSamplesKey) private var favoritedNames = [String]()
+        
+        /// A Boolean value indicating whether the sample is favorited.
+        private var isFavorited: Bool {
+            favoritedNames.contains(String(name.characters))
+        }
+        
+        var body: some View {
+            ZStack {
+                if isFavorited {
+                    HStack {
+                        Image(systemName: "star.fill")
+                            .resizable()
+                            .aspectRatio(contentMode: .fit)
+                            .frame(width: 8)
+                            .padding(.leading, -13)
+                        Spacer()
+                    }
+                }
+                
+                HStack {
+                    VStack(alignment: .leading) {
+                        Text(name)
+                        
+                        if isShowingDescription {
+                            Text(description)
+                                .font(.caption)
+                                .opacity(0.75)
+                                .transition(.move(edge: .top).combined(with: .opacity))
+                        }
+                    }
+                    Spacer()
+                    Label {} icon: {
+                        Image(systemName: "info.circle")
+                            .symbolVariant(isShowingDescription ? .fill : .none)
+                            .imageScale(.medium)
+                    }
+                    .onTapGesture {
+                        isShowingDescription.toggle()
+                    }
+                }
+            }
+            .contextMenu {
+                Button {
+                    if isFavorited {
+                        if let index = favoritedNames.firstIndex(of: String(name.characters)) {
+                            favoritedNames.remove(at: index)
+                        }
+                    } else {
+                        favoritedNames.append(String(name.characters))
+                    }
+                } label: {
+                    if isFavorited {
+                        Label("Unfavorite", systemImage: "star.slash")
+                    } else {
+                        Label("Favorite", systemImage: "star")
+                    }
+                }
+            }
+            .animation(.easeOut(duration: 0.2), value: isShowingDescription)
+        }
+    }
+}
+
+extension SampleLink {
+    /// Bolds given text found in the sample link's label.
+    /// - Parameter text: The text to bold.
+    /// - Returns: A new `SampleLink` with the bolded text.
+    func bolding(_ text: String) -> SampleLink {
```

Comment:
Instead of using a modifier, how about just pass the query as a parameter to the `SampleLink`? A default value can be provided to make the constructor look clean.
I feel it is out of place to pass an variable-length sequence as an argument of a view modifier. Typically it is an option, or a fixed length type.